packages/material-ui/src/useScrollTrigger/useScrollTrigger.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+function getScrollY(ref = window) {
+  if (ref && ref.pageYOffset !== undefined) {
+    return ref.pageYOffset;
+  }
+  if (ref && ref.scrollTop !== undefined) {
+    return ref.scrollTop;
+  }
+  return document
+    ? (document.documentElement || document.body.parentNode || document.body).scrollTop
+    : 0;
+}
+
+function defaultTrigger(event, value, props = {}) {
+  const { directional = true, threshold = 100 } = props;
+  const scrollY = getScrollY(event.currentTarget);
+  // eslint-disable-next-line no-nested-ternary
+  const trigger = directional
+    ? scrollY < value.current
+      ? false
+      : !!(scrollY > value.current && scrollY > threshold)
+    : scrollY > threshold;
+
+  value.current = scrollY;
+  return trigger;
+}
+
+export default function useScrollTrigger(props = {}) {
+  const { onTriggerEval = defaultTrigger, ...initialOptions } = props;
+  const [target, setTarget] = React.useState();
+  const value = React.useRef(0);
```

Comment:
Do you have a more expressive name in mind than a generic value?

Reply:
I switched value to store instead.  I think that is more expressive to it's usage.

Reply:
We have a similar use case in the SwipeableDrawer hook migration effort. I was proposing `instance` for replacing the usage of `this` in a class component. Here, it seems that we store the previous scroll value. I don't have any great name that comes to mind. Using scrollY would conflict with another variable.